turbopack/crates/turbopack-core/src/module_graph/chunk_group_info.rs

```diff
@@ -0,0 +1,345 @@
+                } else {
+                    Some(Either::Left(std::iter::once(ChunkGroup::Entry(
+                        node.module.ident().to_string().await?,
+                    ))))
+                };
+
+                Ok(if let Some(chunk_groups) = chunk_groups {
+                    // let chunk_groups = chunk_groups.collect::<Vec<_>>();
+                    // println!(
+                    //     "{:?} chunk_groups: {:?}",
+                    //     node.module.ident().to_string().await?,
+                    //     chunk_groups
+                    // );
+                    // let chunk_groups = chunk_groups.into_iter();
+
+                    // Start of a new chunk group, don't inherit anything from parent
+                    let chunk_group_ids = chunk_groups.map(|chunk_group| {
+                        if let Some(chunk_group_id) = chunk_groups_to_id.get(&chunk_group) {
+                            chunk_group_id.0
+                        } else {
+                            let chunk_group_id = next_chunk_group_id;
+                            next_chunk_group_id += 1;
+                            chunk_groups_to_id
+                                .insert(chunk_group.clone(), ChunkGroupId(chunk_group_id));
+                            chunk_groups_from_id
+                                .insert(ChunkGroupId(chunk_group_id), chunk_group.clone());
+                            chunk_group_id
+                        }
+                    });
+
+                    // Assign chunk group to the target node (the entry of the chunk group)
+                    let bitset = module_chunk_groups.entry(node.module).or_default();
+                    bitset.0 |= RoaringBitmap::from_iter(chunk_group_ids);
+
+                    // println!(
+                    //     "{:?} chunk_groups: {:?}",
+                    //     node.module.ident().to_string().await?,
+                    //     bitset
+                    //         .iter()
+                    //         .map(|id| chunk_groups_from_id.get(&ChunkGroupId(id)).unwrap())
+                    //         .collect::<Vec<_>>()
+                    // );
+
+                    GraphTraversalAction::Continue
+                } else if let Some((parent, _)) = parent_info {
+                    // Only inherit chunk groups from parent
+
+                    // TODO don't clone here, but both are stored in module_chunk_groups
+                    let parent_chunk_groups =
+                        module_chunk_groups.get(&parent.module).unwrap().clone();
+                    match module_chunk_groups.entry(node.module) {
+                        Entry::Occupied(mut e) => {
+                            // Merge with parent, and continue traversal if modified
+                            let current_chunk_groups = e.get_mut();
+                            if parent_chunk_groups != *current_chunk_groups {
+                                current_chunk_groups.0 |= parent_chunk_groups.0;
```

Comment:
`!=` is the wrong check here. You want to check if `|` will add bits. So you want to check `contains` via `parent_chunk_groups & *current_chunk_groups != *current_chunk_groups`

Reply:
This should be `parent_chunk_groups.is_superset(current_chunk_groups)`
i.e. "there are bits in the parent that are not in current"